main.go

```diff
@@ -1,42 +1,111 @@
 package main
 
 import (
+	"encoding/json"
+	"flag"
 	"fmt"
 	"os"
 	"os/signal"
 	"syscall"
 
+	"github.com/Strum355/log"
 	"github.com/bwmarrin/discordgo"
+	"github.com/hashicorp/consul/api"
+	consulwatch "github.com/hashicorp/consul/api/watch"
 )
 
-func getEnv(key string) (string, error) {
-	val, ok := os.LookupEnv(key)
-	if !ok {
-		return "", fmt.Errorf("Error finding %s", key)
-	}
-	return val, nil
+// Config represents the config pulled from Consul
+type Config struct {
+	PublicServer    string `json:"public"`
+	CommitteeServer string `json:"committee"`
 }
 
+var (
+	config     *Config
+	production bool
+)
+
 func main() {
-	token, err := getEnv("DISCORD_TOKEN")
-	if err != nil {
-		panic(err)
+	config = &Config{}
+	// Check for flags
+	production = *flag.Bool("p", false, "enables production with json logging")
+	flag.Parse()
+	if production {
+		log.InitJSONLogger(&log.Config{Output: os.Stdout})
+	} else {
+		log.InitSimpleLogger(&log.Config{Output: os.Stdout})
 	}
+
+	// Consul config
+	consulConfig := api.DefaultConfig()
+	var err error
+	consulConfig.Address, err = getEnv("CONSUL_ADDRESS")
+	exitError(err)
+
+	readConfig(consulConfig)
+
+	// Discord connection
+	token, err := getEnv("DISCORD_TOKEN")
+	exitError(err)
 	session, err := discordgo.New("Bot " + token)
-	if err != nil {
-		panic("Couldn't init discord connection")
-	}
+	exitError(err)
 	// Open websocket
 	err = session.Open()
-	if err != nil {
-		panic(err)
-	}
+	exitError(err)
 
 	// Maintain connection until a SIGTERM, then cleanly exit
-	fmt.Println("Bot running")
+	log.Info("Bot is Running")
 	sc := make(chan os.Signal, 1)
 	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM, os.Interrupt, os.Kill)
 	<-sc
-	fmt.Println("Cleanly exiting")
+	log.Info("Cleanly exiting")
 	session.Close()
 }
+
+func readConfig(consulConfig *api.Config) {
+	// Connect to consul
+	// Listen for consul updates
+	params := map[string]interface{}{
```

Comment:
https://github.com/hashicorp/consul/blob/master/command/watch/watch.go#L116-L118 We would need a field for the consul token, which would come from env :)

Reply:
I dont believe it needs to be limited to production. Empty string is valid